modules/loadbalancer/templates/nginx_balance.conf.erb

```diff
@@ -18,10 +18,25 @@ server {
 }
 <%- end -%>
 
-map $upstream_http_govuk_request_id $govuk_request_id {
-  default $upstream_http_govuk_request_id;
-  '' $new_uuid;
-}
+# Set GOVUK-Request-Id if not set
+# See http://nginx.org/en/docs/http/ngx_http_perl_module.html
+perl_modules perl/lib;
+perl_set $govuk_request_id '
+  sub {
+    my $r = shift;
+    my $current_header = $r->header_in("GOVUK-Request-Id");
+
+    if (defined $current_header && $current_header ne "") {
+      return $current_header;
+    } else {
+      my $pid = $r->variable("pid");
+      my $msec = $r->variable("msec");
+      my $remote_addr = $r->variable("remote_addr");
+      my $request_length = $r-variable("request_length");
```

Comment:
shouldn't this be:
`my $request_length = $r->variable("request_length");`
assuming it's de-referencing `request_length` from `$r` and not some other tricksy Perl thing?